src/backend/distributed/planner/query_colocation_checker.c

```diff
@@ -269,15 +276,33 @@ WrapRteRelationIntoSubquery(RangeTblEntry *rteRelation)
 	newRangeTableRef->rtindex = 1;
 	subquery->jointree = makeFromExpr(list_make1(newRangeTableRef), NULL);
 
-	/* Need the whole row as a junk var */
-	Var *targetColumn = makeWholeRowVar(newRangeTableEntry, newRangeTableRef->rtindex, 0,
-										false);
+	Relation relation = relation_open(rteRelation->relid, AccessShareLock);
+	int numberOfAttributes = RelationGetNumberOfAttributes(relation);
+
+	int attributeNumber = 1;
+	for (; attributeNumber <= numberOfAttributes; attributeNumber++)
+	{
+		Form_pg_attribute attributeTuple =
+			TupleDescAttr(relation->rd_att, attributeNumber - 1);
+		Var *targetColumn =
+			makeVar(newRangeTableRef->rtindex, attributeNumber, attributeTuple->atttypid,
+					attributeTuple->atttypmod, attributeTuple->attcollation, 0);
+		TargetEntry *targetEntry =
+			makeTargetEntry((Expr *) targetColumn, attributeNumber,
+							strdup(attributeTuple->attname.data), false);
+
+		if (!list_member_int(requiredAttributes, attributeNumber))
+		{
```

Comment:
would be nice to mention why we add `NULL` target  entries, it is not clear to the readers